hazelcast/src/main/java/com/hazelcast/internal/cluster/impl/AbstractJoiner.java

```diff
@@ -313,7 +313,7 @@ private boolean checkMergeTargetIsNotMember(SplitBrainJoinMessage joinMessage) {
 
     private boolean checkClusterStateActiveBeforeMerge(SplitBrainJoinMessage joinMessage) {
         ClusterState clusterState = clusterService.getClusterState();
-        if (clusterState != ClusterState.ACTIVE) {
+        if (!clusterState.isJoinAllowed()) {
```

Comment:
I agree and as I've said it's not that big of a deal and will probably affect lots of code. We can do this later.